NuMPI/Optimization/polonsky_keer.py

```diff
@@ -0,0 +1,235 @@
+"""
+Implementation of constrained conjugate gradient algorithm as described in,
```

Comment:
Again here Polonsky & Keer is contact mechanics specific but NuMPI is more general. What is the origin of this specific CG flavor? Is it actually Polonsky & Keer?

Reply:
They cite Hestenes and a russian book but it is not clear wether they want to say it's the exact algorithm there or just refer to unconstrained CG. I looked in Hestenes and couldn't find their constrained algorithm.